### ISSUE TITLE
is valid with 0 prefix

### ISSUE THREAD

**Issue by nicolasvahidzein** [halt-hammerzeit/libphonenumber-js#90]
First off, let me say i LOVE your library, it's so easy to use and works amazing well. Thank you a million times over.

Not sure what i am doing wrong but i want to know how to handle the use case when a number has a 0 in the national portion like so:

+33 0 6 12 34 56 78 === this validates as valid but it really is not and also it does not get formatted properly 

the correct number is 

+33  6 12 34 56 78 which this time gets returned properly as +33-6-12-34-56-78

with the 0 the format is not applied but funny enough in the as you type function it shows the number properly formatted with the 0 removed.

For now i found a bypass method wher i reconstruct the number like so and it works nicely but i figured i would give you a shout:

```
                                 var tempPhoneNumber = '+' + formatter.country_phone_code + parse(phoneNumber).phone
```

and then:

```
                                tempPhoneNumber = format(tempPhoneNumber, parse(tempPhoneNumber).country, 'International')
				
				var regEx = new RegExp(' ', 'g') //remove the space and replace with "-" dash
				tempPhoneNumber = tempPhoneNumber.replace(regEx, '-')
```

**Comment by halt-hammerzeit** [comment]
Hmm, yeah, this behaviour seems like a bug which isn't really a bug.
Google's `libphonenumber` was originally created to match phone numbers in a non-strict matter meaning that it would rather attempt to parse a strictly speaking invalid phone number than just dismiss it (it's like web browsers are rendering HTML – they prefer to render at least something as long as they can do that).
So, Google's library behaviour is the same:
https://libphonenumber.appspot.com/phonenumberparser?number=%2B330612345678&country=US
Analogous, it doesn't format the phone number in as you type formatter and it also considers this kind of a phone number valid.
I could introduce some kind of a `{ strict: true }` option which would be passed to `parse(number, options)` and `parse()` could return nothing if the `number` is international and contains a national prefix.
Still, from a website owner's perspective it's better to not loose clients just because they don't know how to write their phone numbers right.
All those national prefixes come from the XXth century when mobile phones weren't around and all those national prefixes will go with the older generation. The newer generations won't even know that national prefixes existed when stationary phones go extinct.
So, if a grandma visits your website and orders a kitchen stove then she will still be able to do that even if she inputs her phone number as `+330612345678` (though it's unlikely for a grandma to order something internationally).
She won't get any formatting but I consider it a penalty for not learning how to input international phone numbers correctly.
In `react-phone-number-input` I don't have such an issue because a grandma will likely be inputting a phone number in a local format (which may have been selected by GeoIP).
If your website is for France only then you should use local phone number format.
It will suit grandma.
But the younger generation may not know how to input a local phone number, they will want to do that in international format.
Therefore, I think, the best way is to accept both local and international phone numbers: if the first sign in `+` then switch to international parsing mode; if not – switch to local parsing mode.
This is the behaviour of `react-phone-number-input`
http://halt-hammerzeit.github.io/react-phone-number-input/

So, in summary, this behaviour is exactly what Google's library does, and I could introduce a special `{ strict: true }` flag for parsing but almost anyone would prefer to forgive such input errors not to loose customers.
If you highlight such a phone number as "wrong" the customer may not know what's wrong and may just navigate away from the website (not placing the order).
The formatting is not critical in this case because it doesn't limit functionality.
I guess this sums it up.
